crates/flox-rust-sdk/src/models/environment.rs

```diff
@@ -1,83 +1,764 @@
-    pub async fn flake_attribute(
-        &self,
-    ) -> Result<FlakeAttribute, EnvironmentError<GenerationError<Git>, ParseInstallableError>> {
-        match self {
-            CommonEnvironment::Named(n) => n
-                .flake_attribute(Default::default())
-                .await
-                .map_err(EnvironmentError::Named),
-            CommonEnvironment::Project(p) => p.flake_attribute().map_err(EnvironmentError::Project),
-        }
+/// Struct representing a local environment in a given location and state
+#[derive(Debug)]
+pub struct Environment<S> {
+    /// absolute path to the environment, typically within `<...>/.flox/name`
+    path: PathBuf,
+    /// Access state of the environment
+    ///
+    /// Implementations distinguish whether whe can [Modify] or only [Read] an environment
+    state: S,
+}
+
+impl<S> PartialEq for Environment<S> {
+    fn eq(&self, other: &Self) -> bool {
+        self.path == other.path
+    }
+}
+
+/// Implementation for an environment in any state
+impl<S: State> Environment<S> {
+    /// Get the environment ref for this environment
+    pub fn environment_ref(&self) -> &EnvironmentRef {
+        self.state.environment_ref()
     }
 
-    pub fn system(&self) -> Cow<str> {
-        match self {
-            CommonEnvironment::Named(n) => n.system(),
-            CommonEnvironment::Project(p) => p.system(),
+    /// Turn the environment into a flake attribute,
+    /// a precise url to interact with the environment via nix
+    ///
+    /// ```
+    /// # tokio_test::block_on(async {
+    /// # use flox_rust_sdk::models::environment::{Environment, Read};
+    /// # use std::path::PathBuf;
+    /// # let tempdir = tempfile::tempdir().unwrap();
+    /// # let path = tempdir.path().canonicalize().unwrap().to_string_lossy().into_owned();
+    /// # let system = "aarch64-darwin";
+    ///
+    /// let env = Environment::init(&path, "test_env".parse().unwrap())
+    ///     .await
+    ///     .unwrap();
+    ///
+    /// let flake_attribute = format!("path:{path}/.flox/test_env#.floxEnvs.{system}.default")
+    ///     .parse()
+    ///     .unwrap();
+    /// assert_eq!(env.flake_attribute(system), flake_attribute)
+    ///
+    /// # })
+    /// ```
+    pub fn flake_attribute(&self, system: impl AsRef<str>) -> FlakeAttribute {
+        let flakeref = PathRef {
+            path: self.path.clone(),
+            attributes: Default::default(),
+        }
+        .into();
+
+        let attr_path = ["", "floxEnvs", system.as_ref(), "default"]
+            .try_into()
+            .unwrap(); // validated attributes
+
+        FlakeAttribute {
+            flakeref,
+            attr_path,
+            outputs: Default::default(),
         }
     }
 
-    pub fn packages(&self) -> Value {
-        todo!()
+    /// Path to the environment definition file
+    pub fn flox_nix_path(&self) -> PathBuf {
+        self.path.join("pkgs").join("default").join("flox.nix")
```

Comment:
Or use the top-level flox.nix?

Reply:
The current template uses pkgs/default so just using that is less invasive. I expect this will change when we revisit mkFloxEnv